weather_mv/loader_pipeline/bq.py

```diff
@@ -82,32 +82,44 @@ def __post_init__(self):
                 ds: xr.Dataset = _only_target_vars(open_ds, self.variables)
                 table_schema = dataset_to_table_schema(ds)
 
-        # Create the table in BigQuery
-        try:
-            table = bigquery.Table(self.output_table, schema=table_schema)
-            self.table = bigquery.Client().create_table(table, exists_ok=True)
-        except Exception as e:
-            logger.error(f'Unable to create table in BigQuery: {e}')
-            raise
+        if not self.dry_run:
+            # Create the table in BigQuery
+            try:
+                table = bigquery.Table(self.output_table, schema=table_schema)
+                self.table = bigquery.Client().create_table(table, exists_ok=True)
+            except Exception as e:
+                logger.error(f'Unable to create table in BigQuery: {e}')
+                raise
+        else:
+            logger.debug("Created the BigQuery table")
 
     def expand(self, paths):
         """Extract rows of variables from data paths into a BigQuery table."""
-        (
+        extracted_rows = (
                 paths
                 | 'ExtractRows' >> beam.FlatMap(
                     extract_rows,
                     variables=self.variables,
                     area=self.area,
                     import_time=self.import_time,
                     open_dataset_kwargs=self.xarray_open_dataset_kwargs)
-                | 'WriteToBigQuery' >> WriteToBigQuery(
-                    project=self.table.project,
-                    dataset=self.table.dataset_id,
-                    table=self.table.table_id,
-                    write_disposition=BigQueryDisposition.WRITE_APPEND,
-                    create_disposition=BigQueryDisposition.CREATE_NEVER)
+            )
 
-        )
+        if not self.dry_run:
+            (
+                extracted_rows
+                | 'WriteToBigQuery' >> WriteToBigQuery(
+                        project=self.table.project,
+                        dataset=self.table.dataset_id,
+                        table=self.table.table_id,
+                        write_disposition=BigQueryDisposition.WRITE_APPEND,
+                        create_disposition=BigQueryDisposition.CREATE_NEVER)
+            )
+        else:
+            (
+                extracted_rows
+                | "Log Extracted Rows" >> beam.Map(logger.debug)
```

Comment:
In a separate issue, we need to update the CLIs for all the weather tools s.t. users can set the log level to debug. Would you mind filing a GH issue for this?

Reply:
Created GH issue #135 for the same.